src/stan/lang/generator/generate_constructor.hpp

```diff
@@ -64,28 +66,30 @@ namespace stan {
       o << INDENT2 << "(void) DUMMY_VAR__;  // suppress unused var warning"
         << EOL2;
       o << INDENT2 << "// initialize member variables" << EOL;
-      generate_member_var_inits(prog.data_decl_, o);
+      generate_try(2, o);
+      generate_member_var_inits(prog.data_decl_, 3, o);
       o << EOL;
-      generate_comment("validate, data variables", 2, o);
-      generate_validate_var_decls(prog.data_decl_, 2, o);
-      generate_comment("initialize data variables", 2, o);
-      generate_var_resizing(prog.derived_data_decl_.first, o);
+      generate_comment("validate, data variables", 3, o);
+      generate_validate_var_decls(prog.data_decl_, 3, o);
+      generate_comment("initialize data variables", 3, o);
+      generate_var_resizing(prog.derived_data_decl_.first, 3, o);
       o << EOL;
 
       bool include_sampling = false;
       bool is_var_context = false;
       bool is_fun_return = false;
 
-      // need to fix generate_located_statements
-      generate_located_statements(prog.derived_data_decl_.second,
-                                  2, o, include_sampling, is_var_context,
-                                  is_fun_return);
+      for (size_t i = 0; i < prog.derived_data_decl_.second.size(); ++i)
```

Comment:
Why not refactor this to be a function with an internal loop since you seem to have the same loops elsewhere?